.devcontainer/configure.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""configure script to get build parameters from user."""
+
+from importlib.machinery import SourceFileLoader
+import inspect
+import json
+import os
+import shlex
+import subprocess
+import sys
+
+# pylint: disable=g-import-not-at-top
+try:
+  from shutil import which
+except ImportError:
+  from distutils.spawn import find_executable as which
+# pylint: enable=g-import-not-at-top
+
+CURRENT_DIR = os.path.dirname(
+    os.path.abspath(inspect.getfile(inspect.currentframe())))
+
+DOCKER_IMAGE_SHORTNAME_DICT = {
+    "base": {
+        "latest": "base-bleeding-edge",
+        "stable": "base"
+    },
+    "nvidia": {
+        "latest": "nvidia-bleeding-edge",
+        "stable": "nvidia"
+    }
+}
+
+
+def run_shell(cmd, allow_non_zero=False, stderr=None):
+  cmd = shlex.split(cmd)
+  if stderr is None:
+    stderr = sys.stdout
+  if allow_non_zero:
+    try:
+      output = subprocess.check_output(cmd, stderr=stderr)
+    except subprocess.CalledProcessError as e:
+      output = e.output
+  else:
+    output = subprocess.check_output(cmd, stderr=stderr)
+  return output.decode('UTF-8').strip()
+
+
+def is_nvidia_gpu_available():
+  """This function verifies NVIDIA Docker runtime is installed and 
+  available. It also checks NVIDIA GPU are available and the driver
+  is installed."""
+
+  docker_executable = which('docker')
+  if docker_executable is None:
+    raise RuntimeError("Docker is not installed or in the user path.")
+
+  command_str = f"{docker_executable} info --format '{{{{json .}}}}'"
+  data = json.loads(run_shell(command_str))
```

Comment:
Wrap executable path in quotes in case the path has strings in it.

On Windows, Docker Desktop installs to `C:\ Program Files\Docker\` by default (and it didn't even prompt me to choose another location).

```suggestion
  docker_executable = which('docker')
  if docker_executable is None:
    raise RuntimeError("Docker is not installed or in the user path.")

  command_str = f"'{docker_executable}' info --format '{{{{json .}}}}'"
  data = json.loads(run_shell(command_str))
```